docs/source/core_specification/schema_types.rst

```diff
@@ -1,3 +1,282 @@
+    }s
+
+Inheritance
+-----------
+
+The compiler uses GraphQL interfaces and GraphQL unions in representing the inheritance structure
+of the underlying schema. You may ignore this section if are compiling to a language with no
+inheritance structure, (e.g. SQL).
+
+Interfaces
+~~~~~~~~~~
+
+GraphQL interfaces represent the abstract vertices of the underlying database.
+
+.. code::
+
+    interface Entity {
+        name: String
+        alias: [String]
+        in_Entity_Related: [Entity]
+        out_Entity_Related: [Entity]
+    }
+
+Abstract inheritance is modeled through interface implementation.
+
+.. code::
+
+    type Food implements Entity {
+        name: String
+        alias: [String]
+        in_Entity_Related: [Entity]
+        out_Entity_Related: [Entity]
+        in_Species_Eats: [Species]
+    }
+
+Unions and :code:`type_equivalence_hints`
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+GraphQL does not support a notion of concrete inheritance. In other words, GraphQL objects cannot
+inherit from other objects. However, for certain query dialects, the compiler needs concrete
+inheritance information to emit the right query.
+
+To model concrete inheritance. we use GraphQL unions that encompass an object's
+subclasses and a :code:`type_equivalence_hints` parameter to signify that object is equivalent to
+the GraphQL union.
+
+For example, suppose :code:`Food` and :code:`Species` are concrete types and :code:`Food` is a
+superclass of :code:`Species` in an OrientDB schema. Then the GraphQL schema generation function
+would generate an union class in the schema
+
+.. code::
+
+    union Union__Food__Species = Food | Species
+
+and an entry in :code:`type_equivalence_hints` mapping :code:`Food` to
+:code:`Union_Food_Species`.
+
+
+Type coercions
+~~~~~~~~~~~~~~
+
+Type coercions are operations than can be run against interfaces and unions to create a new scope
+whose type is different than the type of the enclosing scope of the coercion. Type coercions are
+represented with GraphQL inline fragments.
+
+Example Use
+^^^^^^^^^^^
+
+.. code::
+
+    {
+        Species {
+            name @output(out_name: "species_name")
+            out_Species_Eats {
+                ... on Food {
+                    name @output(out_name: "food_name")
+                }
+            }
+        }
+    }
+
+Here, the :code:`out_Species_Eats` vertex field is of the
+:code:`Union__Food__FoodOrSpecies__Species` union type. To proceed with the
+query, the user must choose which of the types in the
+:code:`Union__Food__FoodOrSpecies__Species` union to use. In this example,
+:code:`... on Food` indicates that the :code:`Food` type was chosen, and any
+vertices at that scope that are not of type :code:`Food` are filtered out
+and discarded.
+
+.. code::
+
+    {
+        Species {
+            name @output(out_name: "species_name")
+            out_Entity_Related {
+                ... on Species {
+                    name @output(out_name: "food_name")
```

Comment:
This is not food if you use `out_Entity_Related`. Copy pasta error?

Reply:
Fixed this in the README.rst too.